src/Ocelot/Multiplexer/MultiplexingMiddleware.cs

```diff
@@ -46,11 +46,14 @@ public async Task Invoke(HttpContext httpContext)
             return;
         }
 
+        // WIP: generalize buffer use?
+        var bufferedBody = await BufferBodyAsync(httpContext, route, downstreamRoutes);
```

Comment:
I don't like this idea, why haven't you put the body copy nearer to ```CreateThreadContextAsync```. Do you want to manipulate the request body?

Reply:
@PaulARoy @raman-m I think we should be clear about the purpose of the multiplexer: Create the http context copies (copying bodies etc.), send the requests and then aggregate them. If you want to deal with the buffer, modify the request body etc. then we should think about it somewhere else, not here.

Reply:
@ggnaegi I thought I could not update the body in-place, I 100% aggree with you. The next iteration I just pushed is more fluent and makes the update in-place. It's still in the multiplexer BUT it can be easily moved to a more general approach.

Maybe a new `BufferingMiddleware`? What do you think?

Reply:
@PaulARoy The decision is yours! Please create a backup branch in your local repository to save your work, or stash it. Could you also remove any additional logic that falls outside the scope of the bug? I'm confident that your innovative `BufferingMiddleware`  idea will come to fruition in the subsequent PR.

Reply:
Unfortunately buffering is needed for the functionnality to work.

Explanation: if we don't buffer the body, once the stream is read it cannot be repositioned to 0. The error is pretty non-explicit.

Still if we want to just isolate the basic copy, I can remove the buffering logic and entirely rely on the fact that buffering is manually activated (either with a future custom implementation of the buffering logic, or with the AspNet capability).

The PR becomes pretty small (and I have doubts regarding testability) but that's entirely up to you.

Reply:
> The PR becomes pretty small (and I have doubts regarding testability) but that's entirely up to you.

Well... testing your changes is crucial. However, I can assist you with delivering the bug fix.